cardano-db/src/Cardano/Db/Insert.hs

```diff
@@ -313,6 +315,12 @@ insertCheckOffChainPoolFetchError pofe = do
   foundMeta <- existsPoolMetadataRefId (offChainPoolFetchErrorPmrId pofe)
   when (foundPool && foundMeta) . void $ insertCheckUnique "OffChainPoolFetchError" pofe
 
+insertOffChainVoteData :: (MonadBaseControl IO m, MonadIO m) => OffChainVoteData -> ReaderT SqlBackend m ()
+insertOffChainVoteData pod = void $ insertCheckUnique "OffChainVoteData" pod
```

Comment:
There is a race condition here.
Normally when we insert in a table A that references another table B we do the following
```
bid <- queryTableB hash
insertA $ A {abid = bid, ...}
```
This provides atomicity, the query and the insert happens in the same db transaction. So it's impossible to have a rollback between the queryTableB and insertA

However here we have
```
thread 1:
bid <- queryTableB hash
pushQueue $ A { abid = bid }

thread 2:
a <- flushQueue
insertA a
```

So atomicity is lost, since insert and query don't happen in the same db transaction. This means that a rollback between the 2 operations will cause invalid data to be inserted in the metadata table. The solution is to redo the query inside insertA to validate that all referenced data are still there. You can check how `insertCheckOffChainPoolData` and `insertCheckOffChainPoolFetchError` solve the issue.

There is possibly an alternative solution, to clean the Queue when there is a rollback, but this would need some thinking.